front-end/src/variable/VarDeclaration.h

```diff
@@ -12,15 +12,20 @@ class VarDeclaration : public Declaration {
     std::string name;
     Type type;
     Expr *assignment;
+
 public:
     const std::string &getName() const;
 
     VarDeclaration(const std::string &name, Type type, Expr *assignment = nullptr);
 
+    VarDeclaration(Type type);
```

Comment:
You have changed `VarDeclaration`, but why ? It's `FunctionParam` who need to be changed no ?

Reply:
FunctionParam heritates from VarDeclaration

Reply:
Ok nevermind